org.eclipse.gef/src/org/eclipse/gef/SelectionManager.java

```diff
@@ -289,17 +297,15 @@ public void setSelection(ISelection newSelection) {
 		}
 		selection.clear();
 
-		if (!orderedSelection.isEmpty()) {
-			Iterator itr = orderedSelection.iterator();
-			while (true) {
-				EditPart part = (EditPart) itr.next();
-				selection.add(part);
-				if (!itr.hasNext()) {
-					part.setSelected(EditPart.SELECTED_PRIMARY);
-					break;
-				}
-				part.setSelected(EditPart.SELECTED);
+		Iterator<EditPart> itr = orderedSelection.iterator();
+		while (itr.hasNext()) {
+			EditPart part = itr.next();
+			if (part == null) {
+				LOGGER.error(GEFMessages.SelectionManager_Selection_Is_Null);
+				continue;
```

Comment:
how about adding after the loop an 
```
if(!selection.isEmpty()){
   selection.getLast().setSelected(EditPart.SELECTED_PRIMARY);
}
```

Reply:
After sleeping about it, I'm not quite sure if it's really a good idea to just not throw the exception anymore. I think recent commits already showed that seemingly safe changes can break someones editor.

Instead, it makes more sense to throw the exception before the internal selection is updated. Just so that everything remains consistent.

Reply:
I think the new tests illustrate the desired behavior: The list of selected edit parts should never contain `null`.

Reply:
As I had last week a bug in 4diac IDE where I got an NPE at that point I agree with you. The NPE showed me an important bug in our code which otherwise would have gone unnoticed and with no user feedback. Would have been hard to notice and debug. Therefore I think your proposed improvement goes in the right direction.